cg/meta/deliver/delivery_api.py

```diff
@@ -0,0 +1,245 @@
+                if case_name:
+                    file_name: str = file_name.replace(case.internal_id, case.name)
+                out_path = Path(delivery_base, file_name)
+                if create_link(source=file_path, destination=out_path, dry_run=self.dry_run):
+                    number_linked_files += 1
+
+            if not files and number_linked_files == 0:
+                raise MissingFilesError(f"Could not find any files for sample {sample_id}")
+
+    def _create_sample_delivery_directory(self, case: Case, sample: Sample, pipeline: str) -> Path:
+        case_name: str | None = get_delivery_case_name(case=case, pipeline=pipeline)
+        delivery_base = get_delivery_dir_path(
+            case_name=case_name,
+            sample_name=sample.name,
+            customer_id=case.customer.internal_id,
+            ticket=case.latest_ticket,
+            base_path=self.project_base_path,
+        )
+        if not self.dry_run:
+            LOG.debug(f"Creating project path {delivery_base}")
+            delivery_base.mkdir(parents=True, exist_ok=True)
+        return delivery_base
+
+    def _get_deliverable_samples(self, case: Case, pipeline: str) -> Iterable[CaseSample]:
+        return filter(
+            lambda link: self._sample_is_deliverable(link=link, case=case, pipeline=pipeline),
+            self.store.get_case_samples_by_case_id(case.internal_id),
+        )
+
+    def _get_version_for_sample(self, sample: Sample, case: Case, pipeline: str) -> Version:
+        bundle: str = sample.internal_id if pipeline == DataDelivery.FASTQ else case.internal_id
+        return self.hk_api.last_version(bundle)
+
+    def _get_case_files_from_version(
+        self, version: Version, sample_ids: set[str], pipeline: str
+    ) -> Iterable[Path]:
+        """Fetch all case files from a version that are tagged with any of the case tags."""
+
+        if not version:
+            LOG.warning("Version is None, cannot get files")
+            return []
+
+        if not version.files:
+            LOG.warning(f"No files associated with Housekeeper version {version.id}")
+            return []
+
+        version_file: File
+        for version_file in version.files:
+            if not include_file_case(file=version_file, sample_ids=sample_ids, pipeline=pipeline):
+                LOG.debug(f"Skipping file {version_file.path}")
+                continue
+            yield Path(version_file.full_path)
+
+    def _get_sample_files_from_version(
+        self,
+        version: Version,
+        sample_id: str,
+        pipeline: str,
+    ) -> Iterable[Path]:
+        """Fetch files for a sample from a version that are tagged with sample tags."""
+        file_obj: File
+        for file_obj in version.files:
+            if not include_file_sample(file=file_obj, sample_id=sample_id, pipeline=pipeline):
+                continue
+            yield Path(file_obj.full_path)
```

Comment:
I think this `yield` is confusing. I see no reason why this should not just return `list[Path]`